go-client/client/query.go

```diff
@@ -1441,23 +1483,42 @@ func (op mergeOp) childQueries() []QueryNode {
 func (op mergeOp) makeBatchOp(resultId *ticketpb2.Ticket, children []*tablepb2.TableReference) tablepb2.BatchTableRequest_Operation {
 	assert(len(children) == len(op.children), "wrong number of children for Merge")
 
-	req := &tablepb2.MergeTablesRequest{ResultId: resultId, SourceIds: children, KeyColumn: op.sortBy}
+	var nonNilChildren []*tablepb2.TableReference
+	for _, child := range children {
+		if child != nil {
+			nonNilChildren = append(nonNilChildren, child)
+		}
+	}
+
+	req := &tablepb2.MergeTablesRequest{ResultId: resultId, SourceIds: nonNilChildren, KeyColumn: op.sortBy}
 	return tablepb2.BatchTableRequest_Operation{Op: &tablepb2.BatchTableRequest_Operation_Merge{Merge: req}}
 }
 
 func (op mergeOp) execSerialOp(ctx context.Context, stub *tableStub, children []*TableHandle) (*TableHandle, error) {
 	assert(len(children) == len(op.children), "wrong number of children for Merge")
 
-	return stub.merge(ctx, op.sortBy, children)
+	var nonNilChildren []*TableHandle
+	for _, child := range children {
+		if child != nil {
+			nonNilChildren = append(nonNilChildren, child)
+		}
+	}
+
+	return stub.merge(ctx, op.sortBy, nonNilChildren)
```

Comment:
Line 1532 is actually okay. It is perfectly valid to append a `QueryNode` to a builder that comes from a nil table handle, and `mergeOp` is allowed to have child nodes that come from nil handles because it's checked in `execSerialOp` and `makeBatchOp`.